starter-code/iteration-2/main.ts

```diff
@@ -1,6 +1,41 @@
 // Add the reference to the "TodoInterface"
-
+interface taskTemplate {
+    taskArray : Array <string>;
+    addTask(task : string) : number;
+    listAllTasks() : void;
+    deleteTask(task : string) : number;
+}
 // 1. Create a class Todo that implements the Interface created before.
+class Todo implements taskTemplate {
+    task : string;
+
+    constructor (public taskArray : Array <string>) {} 
```

Comment:
```
class Todo implements taskTemplate {
    task: string;
    taskArray: Array<string> = [];
    constructor() { }

    addTask(task): number {
        this.taskArray.push(task);
        console.log(`=========== NEW TASK ===========
            Task ${task} inserted in the list
            Number of items: ${this.taskArray.length}`);
        return this.taskArray.length;
    }
    ...
}
```

Reply:
You have to give an empty value to the string if you want to push it.

Reply:
Here you are asking for an argument in the constructor:
```constructor (public taskArray : Array <string>) ```
When you call this constructor you are not passing any argument:
``` let myTodos = new Todo();```